mpcs/src/basefold/commit_phase.rs

```diff
@@ -212,3 +227,80 @@ pub(crate) fn basefold_one_round_by_interpolation_weights<
         2,
     )
 }
+
+#[allow(clippy::too_many_arguments)]
+fn basefold_one_round<E: ExtensionField, Spec: BasefoldSpec<E>>(
+    pp: &<Spec::EncodingScheme as EncodingScheme<E>>::ProverParameters,
+    prover_states: &mut Vec<IOPProverState<'_, E>>,
+    challenge: Option<Challenge<E>>,
+    sumcheck_messages: &mut Vec<Vec<E>>,
+    initial_oracle: &[E],
+    transcript: &mut impl Transcript<E>,
+    trees: &mut Vec<MerkleTreeExt<E>>,
+    roots: &mut Vec<<Poseidon2ExtMerkleMmcs<E> as Mmcs<E>>::Commitment>,
+    mmcs_ext: &ExtensionMmcs<
+        E::BaseField,
+        E,
+        <<E as ExtensionField>::BaseField as PoseidonField>::MMCS,
+    >,
+    is_last_round: bool,
+) -> Option<Challenge<E>>
+where
+    E::BaseField: Serialize + DeserializeOwned,
+    <Poseidon2ExtMerkleMmcs<E> as Mmcs<E>>::Commitment:
+        IntoIterator<Item = E::BaseField> + PartialEq,
+{
+    let prove_round_and_update_state_span = entered_span!("prove_round_and_update_state");
+    let prover_msgs = prover_states
+        .par_iter_mut()
+        .map(|prover_state| IOPProverState::prove_round_and_update_state(prover_state, &challenge))
+        .collect::<Vec<_>>();
+    exit_span!(prove_round_and_update_state_span);
+
+    // for each round, we must collect #SIZE prover message
+    let evaluations: AdditiveVec<E> =
+        prover_msgs
+            .into_iter()
+            .fold(AdditiveVec::new(3), |mut acc, prover_msg| {
+                acc += AdditiveVec(prover_msg.evaluations);
+                acc
+            });
+
+    transcript.append_field_element_exts(&evaluations.0);
+    sumcheck_messages.push(evaluations.0);
+
+    let next_challenge = transcript.sample_and_append_challenge(b"commit round");
```

Comment:
shall we also write round index to transcript?

Reply:
I feel it doesn't affect soundness, and to have round index in format string, somehow we need to change to transcript signature since it take &'static str, which can be shift in later scope

Reply:
yeah, we can revisit this later.